### ISSUE TITLE
DirectoryEntry RefreshCache for property subset not clearing uncommitted changes

### ISSUE THREAD

**Issue by hellblade** [dotnet/runtime#87197]
### Description

The DirectoryEntry.RefreshCache method has an overload to take in just a limited number of properties to refresh.
[See Documentation](https://learn.microsoft.com/en-us/dotnet/api/system.directoryservices.directoryentry.refreshcache?view=dotnet-plat-ext-7.0#system-directoryservices-directoryentry-refreshcache(system-string()))

This does not seem to refresh the cache for a property when:
- The object in Acitve Directory does not have any existing values for the property
- You have added a value to the property collection cache

It works correctly if the object does have an existing value for the property, or of course with the non-overloaded RefreshCache method.

### Reproduction Steps

You will need an Active Directory server, and the object guid for an object with no publicDelegates set. Note it's not limited to publicDelegates so could use another too.

```csharp
var directoryEntry = new DirectoryEntry($"LDAP://{server}/<GUID={objectGuid}>", username, password, AuthenticationTypes.Sealing | AuthenticationTypes.Secure | AuthenticationTypes.Signing);
           
directoryEntry.RefreshCache(new string[] { "publicDelegates" });

var delegates = directoryEntry.Properties["publicDelegates"];
		  
// If this GUID is invalid then commit changes throws an error. If it is valid then it gets added
delegates.Add("<GUID=8d80e043-4aaf-4439-9549-acea08998e74>");
directoryEntry.RefreshCache(new string[] { "publicDelegates" });

directoryEntry.CommitChanges();
```

Using the non-overloaded RefreshCache works correctly and the object has no changes made to it (and no exception if the GUID is invalid):
```csharp
var directoryEntry = new DirectoryEntry($"LDAP://{server}/<GUID={objectGuid}>", username, password, AuthenticationTypes.Sealing | AuthenticationTypes.Secure | AuthenticationTypes.Signing);
           
directoryEntry.RefreshCache(new string[] { "publicDelegates" });

var delegates = directoryEntry.Properties["publicDelegates"];
		  
delegates.Add("<GUID=8d80e043-4aaf-4439-9549-acea08998e74>");
directoryEntry.RefreshCache();

directoryEntry.CommitChanges();
```

### Expected behavior

Calling `RefreshCache(String[])` should clear the cache for all properties provided, even if the object bound to has no values for the property.

### Actual behavior

Calling `RefreshCache(String[])` does not clear the cache properties provided where the object bound has no existing values.

### Known Workarounds

Use the `RefreshCache()` overload which may mean more data than required is read in.

### Configuration

.NET6
Windows 11
x64

**Comment by danmoseley** [comment]
Does this repro in .NET Framework?

**Comment by hellblade** [comment]
@danmoseley Yeah, just tried now and it happens with .NET Framework 4.8 also.

**Comment by steveharter** [comment]
The logic looks fairly straighfoward: https://github.com/dotnet/runtime/blob/32440e59f7363f9ce8d31ae40523b09de9bccada/src/libraries/System.DirectoryServices/src/System/DirectoryServices/DirectoryEntry.cs#L956

Basically, remove the property from the underlying dictionary which will then get re-populated lazily when the property is accessed again.

Perhaps try this:
```cs
directoryEntry.Properties["publicDelegates"].Add("<GUID=8d80e043-4aaf-4439-9549-acea08998e74>");
```
instead of
```cs
delegates.Add("<GUID=8d80e043-4aaf-4439-9549-acea08998e74>");
```

**Comment by hellblade** [comment]
Yeah, from that method code nothing stands out. I can see it has removed the value from the underlying dictionary, although the property collection looks to have loaded all properties somehow so it might be another spot affecting it.

`directoryEntry.Properties["publicDelegates"].Add("<GUID=8d80e043-4aaf-4439-9549-acea08998e74>");`
Still seems to have the same result.

Here's just before the Refresh Cache gets called:
![image](https://github.com/dotnet/runtime/assets/3614290/d491d6bf-17f1-48ec-8ca6-3617158fcdc1)

And after, the underlying collection is empty but the overall count of the PropertiesCollection is now all the attributes plus the new one.
![image](https://github.com/dotnet/runtime/assets/3614290/e69beed7-9dbb-4f9a-af34-44d2782ec6db)
